zulipterminal/ui_tools/boxes.py

```diff
@@ -1112,22 +1112,34 @@ def reactions_view(self, reactions: List[Dict[str, Any]]) -> Any:
         if not reactions:
             return ""
         try:
-            reaction_stats = defaultdict(set)
+            my_user_id = self.model.user_id
+            reaction_stats = defaultdict(list)
             for reaction in reactions:
                 user_id = int(reaction["user"].get("id", -1))
                 if user_id == -1:
                     user_id = int(reaction["user"]["user_id"])
-                reaction_stats[reaction["emoji_name"]].add(user_id)
+                user_name = reaction["user"]["full_name"]
+                if user_id == my_user_id:
+                    user_name = "You"
+                reaction_stats[reaction["emoji_name"]].append((user_id, user_name))
 
             sorted_stats = sorted(
                 (reaction, count) for reaction, count in reaction_stats.items()
             )
 
-            my_user_id = self.model.user_id
+            for reaction, ids in sorted_stats:
+                if (my_user_id, "You") in ids:
+                    ids.remove((my_user_id, "You"))
+                    ids.append((my_user_id, "You"))
+
             reaction_texts = [
                 (
-                    "reaction_mine" if my_user_id in ids else "reaction",
-                    f" :{reaction}: {len(ids)} ",
+                    "reaction_mine"
+                    if my_user_id in [id[0] for id in ids]
+                    else "reaction",
+                    f" :{reaction}: {len(ids)} "
+                    if len(ids) > 3 or len(reactions) > 3
```

Comment:
Do we actually need the `len(ids) > 3` condition here?

Also lets define variables for hardcoded constants. Eg: `MAX_USERNAMES_PER_REACTIONS = 3` or something similar.